.github/scripts/lib/release-gate-core.ps1

```
@@ -0,0 +1,99 @@
+}
+
+function Get-ReleaseGateWaiver {
+    param([string]$CommitMessage)
+    if ($CommitMessage -match '(?m)^Skip-Release-Check:\s*all\b') { return 'all' }
+    if ($CommitMessage -match '(?m)^Skip-Release-Check:\s*changelog-only\b') { return 'changelog-only' }
+    return $null
+}
+
+function Invoke-ReleaseGateEvaluation {
+    param(
+        [Parameter(Mandatory)][string[]]$ChangedFiles,
```

PSScriptAnalyzer (1.25.0)

[warning] The parameter 'ChangedFiles' has been declared but not used.

(PSReviewUnusedParameter)